### PR TITLE
update rep-controller image

### PR DESCRIPTION
**Pull request by abhi16394**
# Description
bumping replication-controller image for csi-powerscale- v2.5.0

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/491 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: operatorconfig/moduleconfig/common/version-values.yaml [hunk 1]
```diff
@@ -16,13 +16,10 @@ powerflex:
   # List of Driver versions and modules that supports the version
   v2.3.0:
       authorization: "v1.3.0"
-      replication: "v1.3.0"
```
Comment (bharathsreekanth) [inline operatorconfig/moduleconfig/common/version-values.yaml:19]: why are these removed?

Comment (abhi16394) [reply operatorconfig/moduleconfig/common/version-values.yaml:19]: powerflex doesn't support replication yet 
so removing it, i dont know who added it

Comment (bharathsreekanth) [reply operatorconfig/moduleconfig/common/version-values.yaml:19]: thanks!

**Review by bharathsreekanth — APPROVED** [review]

**Review by harshaatdell — APPROVED** [review]

**Review by kumarkgosa — APPROVED** [review]

File: samples/storage_csm_powerscale_v250.yaml [hunk 2]
```diff
@@ -280,7 +280,7 @@ spec:
       - name: dell-replication-controller-manager
         # image: Defines controller image. This shouldn't be changed
         # Allowed values: string
-        image: dellemc/dell-replication-controller:v1.3.0
+        image: dellemc/dell-replication-controller:v1.3.1
         envs:
```
Comment (rensyct) [inline samples/storage_csm_powerscale_v250.yaml:284]: @abhi16394 , any reason behind not bumping up the config version of replication module

Comment (bharathsreekanth) [reply samples/storage_csm_powerscale_v250.yaml:284]: @rensyct could you give more details on this for our understanding? Where is the config version?

Comment (rensyct) [reply samples/storage_csm_powerscale_v250.yaml:284]: Line No 261 in the same file

Comment (bharathsreekanth) [reply samples/storage_csm_powerscale_v250.yaml:284]: I believe no new config version was added, since there were nothing in there that needed change (although for consistency it would have been nice). https://github.com/dell/csm-operator/tree/main/operatorconfig/moduleconfig/replication has 1.3.0 as the latest version. I am guessing this is the reason for keeping config version as 1.3.0, but I will let @abhi16394 clarify.

Comment (rensyct) [reply samples/storage_csm_powerscale_v250.yaml:284]: We should add new folder for replication under https://github.com/dell/csm-operator/tree/main/operatorconfig/moduleconfig/replication if config version was bumped else validation in code fails

Comment (abhi16394) [reply samples/storage_csm_powerscale_v250.yaml:284]: the replication team is not releasing the entire module
they are bumping up just the replication-controller image
Also there are no changes in replication module folder as the image is in the go file under replication.go
